aeron-driver/src/main/c/aeron_driver_conductor_proxy.c

```diff
@@ -19,31 +19,15 @@
 #include "aeron_alloc.h"
 #include "aeron_driver_conductor.h"
 
-void aeron_driver_conductor_proxy_offer(aeron_driver_conductor_proxy_t *conductor_proxy, void *cmd)
+void aeron_driver_conductor_proxy_offer(aeron_driver_conductor_proxy_t *conductor_proxy, void *cmd, size_t length)
 {
-    while (aeron_mpsc_concurrent_array_queue_offer(conductor_proxy->command_queue, cmd) != AERON_OFFER_SUCCESS)
+    while (aeron_mpsc_rb_write(conductor_proxy->command_queue, 1, cmd, length) < 0)
```

Comment:
Handling AERON_RB_ERROR properly would give some peace of mind here. Currently if one of the proxy_offer functions somehow tries to write a message over the length limit, they end up spinning on this error.

This applies to all the proxy_offer implementations.